### PR TITLE
Speeding up SAMSTS

### PR DESCRIPTION
**Pull request by max-nova**
Didn't manage to reduce the memory profile a ton, but I did cut a TON of overhead.  To chop the memory in half, we'd need to do weird stuff with float16 and avoid underflow.

### CODE REVIEW

**Comment by max-nova** [comment]
The big change is in fill.py - I totally restructured it and used np.save and np.load to save arrays to the filesystem so that I didn't have to lug them around in memory.

File: samsts/cache.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import json
+import logging
+import numpy as np
+from path import Path
+
+CACHE = Path('/mnt/vol/cache').mkdir_p()
+RAW_LS_DL = CACHE.joinpath('landsat-download.json')
```
Comment (mholkesvik) [inline samsts/cache.py]: I'd be tempted to add order info onto these cache vars

```python
CACHE_1_RAW_LS_DL = CACHE.joinpath('landsat-download.json')
CACHE_2_CLOUDCUT = CACHE.joinpath('cloudcut.json')
```

File: samsts/fill.py [hunk 2]
```diff
@@ -30,24 +31,36 @@ def fill_scene(
     requires a cluster_pix_idxs dictionary (see cluster.assign_to_cluster) and
     a cluster_neighbors dictionary (see cluster.find_cluster_neighbors).
     """
+    cache_fn = cache.SCENE_FILL.format(scene=scene)
+    if cache.exists(cache_fn):
+        # TODO dtype
```
Comment (mholkesvik) [inline samsts/fill.py]: dtype check? what's the TODO here?

File: samsts/fill.py [hunk 3]
```diff
@@ -30,24 +31,36 @@ def fill_scene(
...
+        return cache.read(cache_fn)
+
     scene_idxs = stack.get_scene_idxs(scene)
+    scene_idx = scene_idxs[0]
     total_gaps = (
         stack.footprint_mask & np.isnan(stack.matrix[:, :, scene_idxs[0]])
```
Comment (mholkesvik) [inline samsts/fill.py]:
```suggestion
        stack.footprint_mask & np.isnan(stack.matrix[:, :, scene_idx])
```

if we got a var we should use it!

File: samsts/cache.py (hunk 1 above)
Comment (max-nova) [reply samsts/cache.py]: Why?

File: samsts/fill.py (hunk 2 above)
Comment (max-nova) [reply samsts/fill.py]: Ah, holdover from an earlier version.  Removing

**Pull request merged**